cmd/valet_link.go

```diff
@@ -0,0 +1,94 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/fatih/color"
+	"strings"
+
+	"github.com/mitchellh/cli"
+	"trellis-cli/trellis"
+)
+
+type ValetLinkCommand struct {
+	UI      cli.Ui
+	Trellis *trellis.Trellis
+}
+
+func (c *ValetLinkCommand) Run(args []string) int {
+	if err := c.Trellis.LoadProject(); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	var environment string
+
+	switch len(args) {
+	case 0:
+		environment = "development"
+	case 1:
+		environment = args[0]
+	default:
+		c.UI.Error(fmt.Sprintf("Error: too many arguments (expected 0 or 1, got %d)\n", len(args)))
+		c.UI.Output(c.Help())
+		return 1
+	}
+
+	config, ok := c.Trellis.Environments[environment]
+	if !ok {
+		c.UI.Error(fmt.Sprintf("Error: %s is not a valid environment", environment))
+		return 1
+	}
+
+	c.UI.Info(fmt.Sprintf("Linking environment %s...\n", environment))
+
+	for key, site := range config.WordPressSites {
+		c.UI.Info(fmt.Sprintf("Linking site %s...", key))
+
+		canonical, _ := c.Trellis.HostsFromDomain(site.SiteHosts[0].Canonical, environment)
+		app := strings.TrimSuffix(canonical.String(), "." + canonical.TLD)
```

Comment:
I *think* this might just be `canonical.SLD`? ie: `example` from `example.com`

Reply:
```yaml
wordpress_sites:
  third.second.com:
    site_hosts:
      - canonical: third.second.test
    local_path: ../bedrock-third-second
    ssl:
      enabled: true
```

Adding debug messages:
```goland
		canonical, _ := c.Trellis.HostsFromDomain(site.SiteHosts[0].Canonical, environment)
		app := strings.TrimSuffix(canonical.String(), "." + canonical.TLD)

		c.UI.Info(fmt.Sprintf("canonical:  %s", canonical))
		c.UI.Info(fmt.Sprintf("canonical.String():  %s", canonical.String()))

		c.UI.Info(fmt.Sprintf("canonical.TLD:  %s", canonical.TLD))
		c.UI.Info(fmt.Sprintf("canonical.SLD:  %s", canonical.SLD))
		c.UI.Info(fmt.Sprintf("canonical.TRD:  %s", canonical.TRD))

		c.UI.Info(fmt.Sprintf("app:  %s", app))
```

Output:
```
Linking site third.second.com...
canonical:  third.second.test
canonical.String():  third.second.test
canonical.TLD:  test
canonical.SLD:  second
canonical.TRD:  third
app:  third.second
```

The goal of `app := strings.TrimSuffix(canonical.String(), "." + canonical.TLD)` is to remove `.test` from `canonical` while keeping the subdomain part (TRD).

I think we should allow users to use subdomains because valet documented that:

> You can use valet link to serve the same project from multiple (sub)domains. To add a subdomain or another domain to your project run valet link subdomain.app-name from the project folder.
>
> -- https://laravel.com/docs/5.8/valet#serving-sites

Reply:
Ah I didn't think about subdomains. Whatever you think is best then. So what's where works